package.json

```diff
@@ -14,48 +14,54 @@
     "eslint": "eslint --fix \"src/**/*.{ts,tsx}\"",
     "prettier": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
     "test": "vitest run",
-    "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook",
+    "ladle": "ladle serve -p 6006",
+    "build-ladle": "ladle build -o dist-ladle",
+    "tsc": "tsc --noEmit",
+    "tsc -w": "tsc --noEmit --watch",
     "find-unused-code": "ts-prune -p ./tsconfig.json --error --ignore src/stories",
     "find-unused-packages": "depcheck --config=./.depcheckrc.yml"
   },
   "dependencies": {
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-router-dom": "^6.2.2",
-    "react-test-renderer": "^17.0.2"
+    "@ladle/react": "^1.1.2",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.3.0",
+    "react-test-renderer": "^18.2.0"
   },
   "devDependencies": {
-    "@storybook/addon-actions": "^6.4.19",
-    "@storybook/addon-essentials": "^6.4.19",
-    "@storybook/addon-links": "^6.4.19",
-    "@storybook/react": "^6.4.19",
-    "@types/node": "^17.0.21",
-    "@types/react": "^17.0.39",
-    "@types/react-dom": "^17.0.13",
+    "@types/node": "^18.0.0",
+    "@types/react": "^18.0.14",
+    "@types/react-dom": "^18.0.5",
     "@types/react-router-dom": "^5.3.3",
-    "@types/react-test-renderer": "^17.0.1",
-    "@typescript-eslint/eslint-plugin": "^5.13.0",
-    "@typescript-eslint/parser": "^5.13.0",
-    "@vitejs/plugin-react": "^1.2.0",
-    "autoprefixer": "^10.4.2",
-    "cssnano": "^5.1.0",
+    "@types/react-test-renderer": "^18.0.0",
+    "@types/styled-components": "^5.1.25",
+    "@typescript-eslint/eslint-plugin": "^5.29.0",
+    "@typescript-eslint/parser": "^5.29.0",
+    "@vitejs/plugin-react": "^1.3.2",
+    "autoprefixer": "^10.4.7",
+    "cssnano": "^5.1.12",
     "depcheck": "^1.4.3",
-    "eslint": "^8.10.0",
+    "eslint": "^8.18.0",
     "eslint-config-prettier": "^8.5.0",
-    "eslint-plugin-import": "^2.25.4",
-    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jsx-a11y": "^6.6.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "eslint-plugin-react": "^7.29.2",
-    "eslint-plugin-react-hooks": "^4.3.0",
-    "happy-dom": "^2.43.1",
+    "eslint-plugin-react": "^7.30.1",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "happy-dom": "^5.3.1",
     "jest-styled-components": "^7.0.8",
-    "prettier": "^2.5.1",
-    "storybook-builder-vite": "^0.1.17",
+    "prettier": "^2.7.1",
+    "react-error-boundary": "^3.1.4",
+    "stacktracey": "^2.1.8",
     "ts-prune": "^0.10.3",
-    "typescript": "^4.6.2",
-    "vite": "^2.8.6",
-    "vite-plugin-svgr": "^1.0.1",
-    "vitest": "^0.5.9"
+    "typescript": "^4.7.4",
+    "vite": "^2.9.12",
+    "vite-plugin-svgr": "^2.2.0",
+    "vitest": "^0.16.0"
+  },
+  "engines": {
+    "yarn": "USE_NPM",
```

Comment:
By specifying a yarn version of `USE_NPM`, this will result in an error whenever you try to run `yarn <any_command>` in the codebase.

The idea here is if a dev is used to using yarn, this will prevent them from accidentally running `yarn <something>` and generating a `yarn.lock` file.

If developers would like to move to using Yarn, they can specify a valid version here and change the NPM version to `USE_YARN`